src/strings.cpp

```diff
@@ -152,49 +152,56 @@ void SetDParamMaxDigits(size_t n, uint count, FontSize size)
 }
 
 /**
- * Copy \a num string parameters from array \a src into the global string parameter array.
- * @param src  Source array of string parameters.
- * @param num  Number of string parameters to copy.
+ * Copy the parameters from the backup into the global string parameter array.
+ * @param backup The backup to copy from.
  */
-void CopyInDParam(const uint64 *src, int num)
+void CopyInDParam(const span<const StringParameterBackup> backup)
 {
-	for (int i = 0; i < num; i++) SetDParam(i, src[i]);
+	for (size_t i = 0; i < backup.size(); i++) {
+		const auto &value = backup[i];
+		if (std::holds_alternative<std::string>(value)) {
+			_global_string_params.SetParam(i, std::get<std::string>(value));
+		} else {
+			_global_string_params.SetParam(i, std::get<uint64_t>(value));
+		}
```

Comment:
You could use the `std::visit` here, and avoid having to use the `holds_alternative`. I don't know if we already defined any form of `overloaded`, as that is kinda mandatory to keep it readable :) But it would mean it becomes something like:

```
std::visit(overloaded{
            [](auto arg) {
                        _global_string_params.SetParam(i, arg);
            }
}
```

And the compiler will deal with the different types.

No clue if it is an improvement :)

Reply:
The `overloaded` bit isn't even needed. It does look prettier now!